src/html/mod.rs

```diff
@@ -1145,6 +1145,190 @@ pub fn generate(
   Ok(files)
 }
 
+pub fn generate_json(
+  ctx: GenerateCtx,
+) -> Result<HashMap<String, serde_json::Value>, anyhow::Error> {
+  let mut files = HashMap::new();
+
+  // Index page
+  {
+    let (partitions_for_entrypoint_nodes, uses_categories) =
+      if let Some(entrypoint) = ctx.main_entrypoint.as_ref() {
+        let nodes = ctx.doc_nodes.get(entrypoint).unwrap();
+        let categories = partition::partition_nodes_by_category(
+          &ctx,
+          nodes.iter().map(Cow::Borrowed),
+          ctx.file_mode == FileMode::SingleDts,
+        );
+
+        if categories.len() == 1 && categories.contains_key("Uncategorized") {
```

Comment:
Consider cleaning this up later to use `static UNCATEGORIZED: &str = "Uncategorized"`